### DISCUSSION TITLE
Question about ThinLayerHeatTransfer in Heat Conduction Module

### DISCUSSION THREAD

**Discussion by richmondodufisan** [idaholab/moose#25200] [Q&A Modules: General]
"The purpose of this kernel is to model heat transfer across a thin domain". But I have a question on how it works.

The optional term "heat_source" is supposed to define a body heat source if it exists, correct? Just as ADMatHeatSource would do. 

However, if the heat source is on a surface, you would typically apply it as a boundary condition on the surface of the mesh defined by your nodeset/sideset.

How would you do so for the surface of this thin layer? To describe the system, I have a rectangular block which has a thin layer on top of it. The heat is applied to the top of the thin layer, and a boundary conductance is defined for the conductance in between the thin layer and the rectangular block.

If I was doing this with my own mesh (say the layer wasn't so thin), I would have defined a surface for the top of the first block, and define another surface for the interface between both blocks. I could then apply my heat source to the first layer and then define a conductance at the interface.

Well, here, since my mesh does not include the thin layer, I can't define a surface for top of the thin layer. Instead, the top of my mesh is the lower rectangular block, and I already need to use that surface to specify the conductance.

How then, can I define the heat source applied at the surface of the thin layer?

**Comment by lindsayad** [comment]
I believe @jiangwen84 created this object

**Comment by jiangwen84** [comment]
> "The purpose of this kernel is to model heat transfer across a thin domain". But I have a question on how it works.
> 
> The optional term "heat_source" is supposed to define a body heat source if it exists, correct? Just as ADMatHeatSource would do.
> 

Yes, the heat_source term is for the volumetric heat source. 

> However, if the heat source is on a surface, you would typically apply it as a boundary condition on the surface of the mesh defined by your nodeset/sideset.
> 

through the heat flux? 

> How would you do so for the surface of this thin layer? To describe the system, I have a rectangular block which has a thin layer on top of it. The heat is applied to the top of the thin layer, and a boundary conductance is defined for the conductance in between the thin layer and the rectangular block.
> 
> If I was doing this with my own mesh (say the layer wasn't so thin), I would have defined a surface for the top of the first block, and define another surface for the interface between both blocks. I could then apply my heat source to the first layer and then define a conductance at the interface.
> 

If you explicitly mesh the thin layer, I do not think you need to use ```ThinLayerHeatTransfer```. You can just assign the properties to the thin layer block. 

> Well, here, since my mesh does not include the thin layer, I can't define a surface for top of the thin layer. Instead, the top of my mesh is the lower rectangular block, and I already need to use that surface to specify the conductance.
> 

If you do not want to mesh the thin layer (no block defined for the thin layer), you probably can follow ```ThinLayerHeatTransfer``` to implement a thin layer heat transfer BC which only models the one side the the domain (v.s two blocks with interface inside ThinLayerHeatTransfer)

> How then, can I define the heat source applied at the surface of the thin layer?

**Reply by richmondodufisan** [reply]
> through the heat flux?

Yes, the external heat source would be applied as a heat flux BC.

Your suggestion sounds right, but I am not sure how to implement it given the syntax in the documentation. Where would I define the BC, i.e, the applied heat source? For example, see the interface kernel block below:

```
[InterfaceKernels]
  [thin_layer]
    type = ThinLayerHeatTransfer
    thermal_conductivity = \#Thermal Conductivity of the Thin Layer
    specific_heat = \#Specific Heat Capacity of the Thin Layer
    density = \#Density of the Thin Layer
    heat_source = \#Volumetric Heat Source- in my case, leave this option out
    thickness = \#Thickness of the Thin Layer
    variable = "temp" \#Temperature, i.e the same variable used for the rest of the bulk
    neighbor_var = "temp" \#Temperature, i.e the variable used for the rest of the bulk
    boundary = "top_surface" \#The Top surface of my mesh
  []
[]
```

In the above example, I have used the "top_surface" physical group to define where to put the thin layer. Now, if I wanted to apply my heat flux BC (i.e, the applied heat source), I cannot use "top_surface" again. i.e:

```
  [heat_source_term]
    type = FunctionNeumannBC
	variable = temp
	boundary = \#Top of the Thin Layer. 
	function = heat_source_function
  []
```

I tried setting `boundary` to `"thin_layer"` since that was the name of my interface kernel block, but that didn't work. How do I tell MOOSE to apply the Heat Source to the top of the thin layer?
